### PR TITLE
DM-7066: Port to Python 3

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: tests/testBlockTimingLog.py [hunk 1]
```diff
@@ -4,20 +4,17 @@
 """
 from __future__ import with_statement
 
-import pdb                              # we may want to say pdb.set_trace()
-import os
-import sys
+# import pdb                              # we may want to say pdb.set_trace()
```
Comment (PaulPrice) [inline tests/testBlockTimingLog.py]: Kill.

File: tests/testNoDebug.py [hunk 2]
```diff
@@ -28,17 +28,18 @@
...
 
-import sys, os
+import sys
+import os
 
 testdir = os.path.dirname(sys.argv[0])
 if len(testdir) > 0:
     sys.path.insert(0, testdir)
-import verifyOutput
+import verifyOutput  # noqa
```
Comment (PaulPrice) [inline tests/testNoDebug.py]: Is used a couple of lines later; no need for `noqa`?

File: .gitignore [hunk 3]
```diff
@@ -16,6 +16,7 @@ doc/*.tag
 doc/*.inc
 doc/doxygen.conf
 tests/.tests
+tests/.cache
```
Comment (PaulPrice) [inline .gitignore:19]: Squash into previous commit?

Comment (timj) [inline .gitignore:19]: Squashed into the py.test commit.

**Comment by timj** [comment]
@ktlim I've done all the requested changes but I also went crazy and converted the remaining tests so they can run with unittest.

**Pull request merged**